masonry/src/widget/flex.rs

```diff
@@ -518,27 +572,49 @@ impl Widget for Flex {
         // these two are calculated but only used if we're baseline aligned
         let mut max_above_baseline = 0f64;
         let mut max_below_baseline = 0f64;
-        let mut any_use_baseline = self.cross_alignment == CrossAxisAlignment::Baseline;
+        let mut any_use_baseline = false;
+
+        // indicates that the box constrains for the following children have changed. Therefore they
+        // have to calculate layout again.
+        let bc_changed = self.old_bc != *bc;
+        let mut any_changed = bc_changed;
+        self.old_bc = *bc;
 
         // Measure non-flex children.
         let mut major_non_flex = 0.0;
         let mut flex_sum = 0.0;
         for child in &mut self.children {
             match child {
                 Child::Fixed { widget, alignment } => {
-                    any_use_baseline &= *alignment == Some(CrossAxisAlignment::Baseline);
+                    // The BoxConstrains of fixed-children only depends on the BoxConstrains of the
+                    // Flex widget.
+                    let child_size = if bc_changed || ctx.child_needs_layout(widget) {
+                        let alignment = alignment.unwrap_or(self.cross_alignment);
+                        any_use_baseline |= alignment == CrossAxisAlignment::Baseline;
+
+                        let old_size = ctx.widget_state.layout_rect().size();
+                        let child_bc = self.direction.constraints(&loosened_bc, 0.0, 9000.0); // TODO Infinity does lead to NaN and then crashes the app, if this is too big, it leads to floating point issues
```

Comment:
I think I remember discussions about this, to avoid crashes (`NaN` and the likes), I set it to a high but not too high number (as vello goes crazy than, with weird floating point artifacts), the issue also exists for the current `layout` method

Reply:
> to avoid crashes (NaN and the likes), I set it to a high but not too high number

Pretty sure there's [a Codeless Code fable](http://thecodelesscode.com/case/115) about this. :smiley: 

The fundamental problem is that Masonry's layout model is confused, not that Flex had the wrong maximum constraint. Though I guess this works in the meantime.